src/test/java/bit/javaoop/CompanyTest.java

```diff
@@ -0,0 +1,69 @@
+package bit.javaoop;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class CompanyTest {
+
+    @Test
+    void shouldGiveCorrectCompanyHierarchy() {
+        //given
+        Ceo ceo = new Ceo("Tomasz");
+
+        AbstractManager fixedBudgetManager = new FixedBudgetManager("Andrzej",40_000,20_000);
+        AbstractManager fixedEmployeeNumberManager = new FixedEmployeeNumberManager("Ania",40_000,5);
+
+        fixedBudgetManager.addEmployee(
+                new Employee("Arnold", 10_000)
+        );
+        fixedBudgetManager.addEmployee(
+                new Employee("Wojciech", 10_000)
+        );
+        fixedBudgetManager.addEmployee(
+                new Employee("Julia", 10_000)
+        );
+
+        fixedEmployeeNumberManager.addEmployee(
+                new Employee("Kasia", 20_000)
+        );
+        fixedEmployeeNumberManager.addEmployee(
+                new Employee("Basia", 20_000)
+        );
+        fixedEmployeeNumberManager.addEmployee(
+                new Employee("Aleksander", 20_000)
+        );
+
+        ceo.addManager(fixedBudgetManager);
+        ceo.addManager(fixedEmployeeNumberManager);
+
+        Company company = new Company("Kompania Braci", ceo);
+
+        //when
+        String result = company.toString();
+
+        //then
+        assertEquals(getResultPattern(), result);
+
+
+    }
+
+    private String getResultPattern() {
+        StringBuilder resultPattern = new StringBuilder();
+        resultPattern.append("Tomasz - CEO\n");
+        resultPattern.append("\tAndrzej - Manager\n");
+        resultPattern.append("\t\tArnold - Employee\n");
+        resultPattern.append("\t\tWojciech - Employee\n");
+        resultPattern.append("\t\tJulia - Employee\n");
+
+        resultPattern.append("\tAnia - Manager\n");
+        resultPattern.append("\t\tKasia - Employee\n");
+        resultPattern.append("\t\tBasia - Employee\n");
+        resultPattern.append("\t\tAleksander - Employee\n");
+
+        return resultPattern.toString();
+    }
+
+}
```

Comment:
Ja bym to napisał tak:
![image](https://user-images.githubusercontent.com/18233796/38460447-2bb4501e-3aba-11e8-88b3-d4e51a6b390b.png)

- Niewiem dlaczego zrobiło Ci się takie śmieszne formatowanie tych add employee ale bez tylu enterów jest przejrzyjściej:) takie entery się dorzuca jak naprawdę dużo argumentów musimy przekazać do parametrów metody.
- Ten resultPattern jednak wrzuciłbym do testu zamiast z metody faktycznie normalnie w kodzie wywalilibyśmy coś takiego do osobnej metody jednak tutaj to jest ściśle związane z jednym testem a nazwa metody getResultPattern nie mówi nam do którego testu jest ten resultPattern. Jakbyśmy mieli więcej testów i więcej takich metod zrobiłoby się zamieszanie a to totalnie pasuje do bloku //given